src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -234,55 +230,53 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
     //             There should only ever by one of each command (identified by the actionID) in the nameMap.
     void ActionMap::_PopulateNameMapWithStandardCommands(std::unordered_map<hstring, Model::Command>& nameMap) const
     {
-        std::unordered_set<InternalActionID> visitedActionIDs;
-        for (const auto& cmd : _GetCumulativeActions())
+        for (const auto& [_, cmd] : _CumulativeActionMapCache)
         {
-            // only populate with valid commands
-            if (cmd.ActionAndArgs().Action() != ShortcutAction::Invalid)
+            const auto& name{ cmd.Name() };
+            if (!name.empty())
             {
-                // Only populate NameMap with actions that haven't been visited already.
-                const auto actionID{ Hash(cmd.ActionAndArgs()) };
-                if (visitedActionIDs.find(actionID) == visitedActionIDs.end())
-                {
-                    const auto& name{ cmd.Name() };
-                    if (!name.empty())
-                    {
-                        // Update NameMap.
-                        nameMap.insert_or_assign(name, cmd);
-                    }
-
-                    // Record that we already handled adding this action to the NameMap.
-                    visitedActionIDs.emplace(actionID);
-                }
+                // Update NameMap.
+                nameMap.insert_or_assign(name, cmd);

```

Comment:
There is an issue here - previously `cmd.Name` was used for disambiguation between commands, but now we use `cmd.ID`. This means that if there's a command with the same name as another, one of them will overwrite the other